src/jdk.jdwp.agent/share/native/libjdwp/threadControl.c

```diff
@@ -253,8 +253,15 @@ findThread(ThreadList *list, jthread thread)
         if ( list == NULL || list == &otherThreads ) {
             node = nonTlsSearch(getEnv(), &otherThreads, thread);
         }
-        /* A thread with no TLS should never be in the runningThreads list. */
-        JDI_ASSERT(!nonTlsSearch(getEnv(), &runningThreads, thread));
+        /*
+         * Search runningThreads list. The TLS lookup may have failed because the
+         * thread has terminated, but the ThreadNode may still be present.
+         */
+        if ( node == NULL ) {
```

Comment:
nit - s/( /(/ and s/ )/)/

Reply:
The extra spaces are intentional to remain consistent with the surrounding code.